symphony/bdk/core/client/api_client_factory.py

```diff
@@ -76,11 +77,17 @@ def _get_api_client(self, server_config, context) -> ApiClient:
             ApiClientFactory._configure_proxy(server_config, configuration)
 
         client = ApiClient(configuration=configuration)
+        ApiClientFactory._add_headers(client, server_config)
+
+        return client
+
+    @staticmethod
+    def _add_headers( client, server_config):
         for header_name, header_value in ApiClientFactory._sanitized_default_headers(server_config).items():
             client.set_default_header(header_name, header_value)
         client.user_agent = ApiClientFactory._user_agent(server_config)
 
-        return client
+        client._ApiClient__call_api = add_x_trace_id(client._ApiClient__call_api)
```

Comment:
should we check that the X-Trace-Id is not already set by the lines above (where you can customize header) before adding our own?
just in case a customer is hardcoding a X-Trace-Id using the config